cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/config/HazelcastInstanceConfiguration.java

```diff
@@ -0,0 +1,125 @@
+package org.jasig.cas.ticket.registry.config;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MaxSizeConfig;
+import com.hazelcast.config.MulticastConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.config.TcpIpConfig;
+import com.hazelcast.config.XmlConfigBuilder;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.util.ResourceUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Spring's Java configuration component for <code>HazelcastInstance</code> that is consumed and used by
+ * <code>HazelcastTicketRegistry</code>.
+ *
+ * This configuration class has the smarts to choose the configuration source for the <code>HazelcastInstance</code> bean
+ * that it produces by either loading the native hazelcast XML config file from a resource location indicated by
+ * <i>hz.config.location</i> property or if that property is not set nor a valid location, creates HazelcastInstance programattically
+ * with a handful properties and their defaults (if not set) that it exposes to CAS deployers.
+ *
+ * @author Dmitriy Kopylenko
+ * @since 4.2.0
+ */
+@Configuration
+public class HazelcastInstanceConfiguration {
+
+    @Autowired
+    private HazelcastProperties hazelcastProperties;
+
+    /**
+     * Create HazelcastInstance bean.
+     *
+     * @param hazelcastConfigLocation String representation of hazelcast xml config.
+     * @param resourceLoader resource loader for loading hazelcast xml configuration resource.
+     * @return HazelcastInstance bean.
+     *
+     * @throws IOException if parsing of hazelcast xml configuration fails
+     */
+    @Bean
+    public HazelcastInstance hazelcast(@Value("${hz.config.location:NO_CONFIG_PROVIDED}") final String hazelcastConfigLocation,
```

Comment:
oK. I think I get it: the `ResourceLoader` returns `false` to `exists()` in that case, right?